src/parser.rs

```diff
@@ -88,32 +82,23 @@ fn search_for_attr(
     exact_depth: Option<usize>,
 ) -> Result<Vec<(NixNode, String, usize)>, String> {
     // assuming that the root node is an attrset
+    // TODO if it's not, we should fail louder
     let mut stack = match AttrSet::cast((*root_node).clone()) {
-        Some(rn) => rn.entries().map(|x| (x, String::new())).collect(),
+        Some(rn) => rn.entries().map(|x| (x, String::new(), 0)).collect(),
         None => {
             println!("failed to convert!");
             Vec::new()
         }
     };
 
-    let mut remaining_items_prev = stack.len();
-    let mut remaining_items_cur = 0;
-    let mut cur_depth = 0;
     let mut result = Vec::new();
 
     while !stack.is_empty() {
-        let (cur_node, mut path) = stack.pop().unwrap();
+        let (cur_node, mut path, cur_depth) = stack.pop().unwrap();
```

Comment:
This can be made more efficient using just:
```rust
    while let (cur_node, mut path, cur_depth) = stack.pop() {
```

Reply:
:+1: done

Reply:
@DieracDelta you can also just mark such simple suggestions as resolved instead of replying "done".